packages/editor/src/services/ComponentShelfCategoriesState.ts

```diff
@@ -112,27 +103,23 @@ export const ComponentShelfCategoriesState = defineState({
         DirectionalLightComponent,
         HemisphereLightComponent
       ],
-      FX: [
-        LoopAnimationComponent,
-        ShadowComponent,
-        ParticleSystemComponent,
-        EnvmapComponent,
-        SDFComponent,
-        PostProcessingComponent,
-        ReflectionProbeComponent
-      ],
-      Scripting: [VisualScriptComponent],
+      FX: [LoopAnimationComponent, ShadowComponent, ParticleSystemComponent, EnvmapComponent, PostProcessingComponent],
+      Scripting: [],
       Settings: [SceneSettingsComponent, RenderSettingsComponent, MediaSettingsComponent, CameraSettingsComponent],
-      Visual: [
-        EnvMapBakeComponent,
-        ScenePreviewCameraComponent,
-        SkyboxComponent,
-        SplineTrackComponent,
-        SplineComponent,
-        TextComponent,
-        ScreenshareTargetComponent,
-        LookAtComponent
-      ]
+      Visual: [EnvMapBakeComponent, ScenePreviewCameraComponent, SkyboxComponent, TextComponent, LookAtComponent]
     } as Record<string, Component[]>
+  },
+  reactor: () => {
+    const [visualScriptPanelEnabled] = useFeatureFlags([FeatureFlags.Studio.Panel.VisualScript])
+    const cShelfState = getState(ComponentShelfCategoriesState)
```

Comment:
Could we do this with a getMutableState just in case we want to react to this change elsewhere?